packages/realm/src/Counter.ts

```diff
@@ -84,7 +88,7 @@ export class Counter {
 
   /** @internal */
   constructor(realm: Realm, obj: binding.Obj, columnKey: binding.ColKey) {
-    if (!(realm instanceof Realm) || !(obj instanceof binding.Obj)) {
+    if (!(obj instanceof binding.Obj)) {
```

Comment:
is there a reason we're removing this check?

Reply:
I didn't want the file to depend on `Realm` as that would introduce a circular dependency. I think this check is sufficiently strong as it's mainly guarding against misuse of the constructor, from outside of the SDK.